_plugins/getpub.rb

```diff
@@ -98,6 +98,7 @@ def inspire(pub)
       pub['title'] ||= data.dig('titles', 0, 'title')
       pub['link'] ||= "http://inspirehep.net/record/#{recid}"
       pub['date'] ||= data['preprint_date']
+      pub['date'] ||= data.dig('imprints', 0, 'date')
```

Comment:
Some inspire pubs don't have a preprint_date.

Reply:
seems like a line of code I need to clone elsewhere... (but googling dig is not so useful)

But can the line above be removed instead?

Reply:
Dig means try to do `data['imprints'][0]['date']` but return nil if any one of those fails.

If the first line succeeds, then `pub['date']` is no longer nil, and then the `||=` will not change it - it only changes the thing on the left if the thing on the left is falsy (which, in Ruby, is only false and nil).

So, this will prioritize `preprint_date`, but will use `['imprints'][0]['date']` if that is not set.

Reply:
Added a note.

Reply:
@davidlange6 This line is the one that fixes the empty `()` on some pubs.